test/test-lint.cpp

```diff
@@ -1148,6 +1148,23 @@ TEST(test_lint, assign_to_undeclared_variable) {
                             span_matcher(assignment))));
 }
 
+TEST(test_lint, use_and_assign_to_undeclared_variable) {
```

Comment:
Also test the following:
```javascript
x += 1;
let x = 2;
```

Reply:
Added, but because of the scope of project, i moved that test to `test-lint-parse.cpp`.

Reply:
Added, but because of the project structure, i decided to add at `test-lint-parse.cpp`